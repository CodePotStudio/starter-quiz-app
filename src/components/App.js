import React, { useState } from "react";
import { ThemeProvider } from "styled-components";
import theme from "theme";
import GlobalStyle from "globalStyle";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Quiz, Landing, Result, Loading } from "pages";
import { QUIZZES } from "../constants";
import { Helmet } from "react-helmet";
import favicon from "assets/images/favicon.ico";
import elon from "assets/images/elon.jpg";

function App() {
	const [score, setScore] = useState(0);
	const convertedScore = Math.floor((score / QUIZZES.length) * 100);
	const currentUrl = window.location.href;

	return (
		<ThemeProvider theme={theme}>
			<Helmet>
				<title>일론 머스크 지수 테스트하기</title>
				<link rel="icon" href={favicon} />
				<meta property="og:url" content={currentUrl} />
				<meta property="og:title" content="일론 머스크 지수 테스트하기" />
				<meta
					property="og:description"
					content="나는 일론 머스크에 대해서 얼마나 알고 있을까?"
				/>
				<meta property="og:image" content={elon} />
			</Helmet>
			<GlobalStyle />
			<Router>
				<Route path="/result/:cCode">
					<Result convertedScore={convertedScore} setScore={setScore} />
				</Route>
				<Route path="/quiz">
					<Quiz setScore={setScore} />
				</Route>
				<Route path="/loading">
					<Loading convertedScore={convertedScore} />
				</Route>
				<Route path="/" exact>
					<Landing />
				</Route>
			</Router>
		</ThemeProvider>
	);
}

export default App;
