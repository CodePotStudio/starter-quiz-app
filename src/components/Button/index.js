import React from "react";
import styled from "styled-components";

const StyledButton = styled.button`
  font-size: ${(props) => (props.fontSize === "big" ? "32px" : "16px")};
	color: #ffffff;
  background-color: ${(props) => props.theme.primaryColor100};
	border-radius: 5px;
	border: 0px;
	height: 56px;
	padding: 4px;
	margin: 4px;
	cursor: pointer;
	width: 100%;
	outline: none;
	font-weight: 700;
	&:hover {
    background-color: ${(props) => props.theme.primaryColor80};
	}
}
`;

const Button = (props) => (
	<StyledButton onClick={props.onClick}>{props.text}</StyledButton>
);
export default Button;
