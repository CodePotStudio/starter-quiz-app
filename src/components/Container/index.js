import React from "react";
import styled from "styled-components";

const FlexBox = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
`;
const ContainerWrapper = styled.div`
	width: 400px;
	margin-top: 72px;
`;

const Container = (props) => (
	<FlexBox>
		<ContainerWrapper>{props.children}</ContainerWrapper>
	</FlexBox>
);
export default Container;
