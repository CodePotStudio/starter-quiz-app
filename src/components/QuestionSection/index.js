import React from "react";
import styled from "styled-components";
import { QUIZZES } from "../../constants";

const QuestionSectionWrapper = styled.div`
	margin-bottom: 16px;
`;

const PageLabel = styled.h1`
	font-size: 16px;
	font-weight: bold;
	margin-bottom: 8px;
`;

const QuestionTitle = styled.div`
	font-size: 20px;
	margin-bottom: 8px;
`;

const QuestionSection = ({ currentNo }) => (
	<QuestionSectionWrapper>
		<PageLabel>
			<span>{QUIZZES[currentNo].id}</span>/{QUIZZES.length}
		</PageLabel>
		<QuestionTitle>{QUIZZES[currentNo].question}</QuestionTitle>
	</QuestionSectionWrapper>
);
export default QuestionSection;
