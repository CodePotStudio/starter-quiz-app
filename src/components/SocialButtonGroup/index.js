import styled from "styled-components";
import { FacebookShareButton, FacebookIcon } from "react-share";
import { TwitterShareButton, TwitterIcon } from "react-share";
import { CopyToClipboard } from "react-copy-to-clipboard";

// 버튼별로 40px을 확보하고, 각 영역 간에는 8px의 간격을 둠.
const GridContainer = styled.div`
	display: grid;
	grid-template-columns: repeat(4, 48px);
	grid-column-gap: 8px;
	justify-content: center;
	margin-bottom: 16px;
`;

const URLShareButton = styled.button`
	width: 48px;
	height: 48px;
	color: white;
	border-radius: 24px;
	border: 0px;
	font-weight: 800;
	font-size: 18px;
	cursor: pointer;
	background-color: ${(props) => props.theme.primaryColor100};
	&:hover {
		background-color: ${(props) => props.theme.primaryColor80};
	}
`;

const SocialButtonGroup = () => {
	const currentUrl = window.location.href;

	return (
		<GridContainer>
			<FacebookShareButton url={currentUrl}>
				<FacebookIcon size={48} round={true} borderRadius={24}></FacebookIcon>
			</FacebookShareButton>
			<TwitterShareButton url={currentUrl}>
				<TwitterIcon size={48} round={true} borderRadius={24}></TwitterIcon>
			</TwitterShareButton>
			<CopyToClipboard text={currentUrl}>
				<URLShareButton>URL</URLShareButton>
			</CopyToClipboard>
			<button>4</button>
		</GridContainer>
	);
};

export default SocialButtonGroup;
