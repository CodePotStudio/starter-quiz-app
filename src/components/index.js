export { default as AnswerGroup } from "components/AnswerGroup";
export { default as Button } from "components/Button";
export { default as Container } from "components/Container";
export { default as QuestionSection } from "components/QuestionSection";
export { default as ResultSection } from "components/ResultSection";
export { default as Spinner } from "components/Spinner";
export { default as SocialButtonGroup } from "components/SocialButtonGroup";
