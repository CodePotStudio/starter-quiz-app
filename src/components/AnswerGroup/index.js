import React from "react";
import styled from "styled-components";
import { QUIZZES } from "../../constants";
import Button from "../Button";

const AnswerGroupWrapper = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	margin-bottom: 8px;
`;

const AnswerGroup = ({ currentNo, handleClick }) => (
	<AnswerGroupWrapper>
		{QUIZZES[currentNo].answers.map((answer) => (
			<Button
				key={answer.text}
				text={answer.text}
				onClick={() => handleClick(answer.isCorrect)}
			></Button>
		))}
	</AnswerGroupWrapper>
);
export default AnswerGroup;
