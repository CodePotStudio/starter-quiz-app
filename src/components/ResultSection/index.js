import { SocialButtonGroup } from "components";
import React from "react";
import styled from "styled-components";
import RESULT from "./result";

const SectionTitle = styled.h1`
	font-size: 64px;
	font-weight: bold;
	margin-top: 0px;
	margin-bottom: 56px;
	text-align: center;
`;

const StyledImage = styled.img`
	border-radius: 16px;
	opacity: 80%;
	margin-bottom: 16px;
	max-width: 100%;
	display: block;
`;

const ResultTitle = styled.div`
	font-size: 24px;
	margin-bottom: 56px;
	text-align: center;
`;

const ResultSection = ({ cCode }) => (
	<>
		<SectionTitle>당신은 혹시?</SectionTitle>
		<StyledImage src={RESULT[cCode].src} />
		<ResultTitle>{RESULT[cCode].title}</ResultTitle>
		<SocialButtonGroup></SocialButtonGroup>
	</>
);
export default ResultSection;
