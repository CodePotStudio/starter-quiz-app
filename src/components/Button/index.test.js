// components/Button/index.test.js

import { Button } from "components";
import renderer from "react-test-renderer";

test("Hello 버튼 컴포넌트 테스트", () => {
	// 테스트를 실행하면 스냅샷을 생성함
	const tree = renderer
		.create(<Button onClick={() => console.log("hello")}>Hello</Button>)
		.toJSON();
	// 기존의 스냅샷과 비교하여 다른 점이 있는지 파악함
	expect(tree).toMatchSnapshot();
});
