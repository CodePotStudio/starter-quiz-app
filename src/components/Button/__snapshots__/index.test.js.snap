// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 버튼 컴포넌트 테스트 1`] = `
<button
  className="sc-dlfnbm fBmWev"
  onClick={[Function]}
>
  Hello
</button>
`;
